internal/agent/device/systeminfo/system_info.go

```diff
@@ -564,6 +564,16 @@ func getSystemInfoMap(ctx context.Context, log *log.PrefixLogger, info *Info, in
 		infoMap[key] = val
 	}
 
+	for key, collectorfn := range collectors {
+		_, alreadyExists := infoMap[key]
+		if alreadyExists {
+			log.Debugf("SystemInfo collector already populated: %s is %s", key, infoMap[key])
```

Comment:
IIUC, this is a user-defined collector trying to override a key whose value the agent already determined?  If so, we should allow the user to override it in case the agent isn't collecting the information correctly for the given hardware.

Reply:
That's a good insight, @hexfusion what do you think?

Reply:
The contract is that root keys in systemInfo are system-generated, best effort. In the case use wants to provide another mechanism it can be declared on the filesystem as part of customInfo. In this case this would be considered a system-level collector. Docs will reflect.